### PR TITLE
[REVIEW] Add Dask-RF notebook

### PR DESCRIPTION
**Pull request by JohnZed**
I'll clear out the outputs before checking in, but figured it couldn't hurt to show them for review.

### CODE REVIEW

**Comment by JohnZed** [comment]
@cjnolet could you take a look?

**Comment by cjnolet** [comment]
Lookling at this now

**Comment by cjnolet** [comment]
It's hard to go through the diffs for notebooks so I'm just going to leave some feedback in a comment. Overall, fairly minor. Here's my thoughts:

- I see unused imports 
- Can we make the subheadings 2-levels deep rather than top-level? (eg ## instead of #)? That's more consistent w/ the other notebooks. 
- To make it blatantly obvious to the users where the device and host memory data are being generated, I've been separating them into two different cells, with third level headings for "Host" and "Device". Just makes it easier to follow the flow of data for sklearn vs data for cuml. 
- Maybe rather than repeating the docs, we should link to dask-cudf and cuML RF docs at the end of the description at the top. This lowers the chance that the notebook will need to be modified after each release. For instance, if we fix some of these issues in the next release, we can just update the docs.

What do you think?

**Comment by JohnZed** [comment]
I implemented the suggestions listed above. I still don't love that we use `##` instead of `#` for headings, but I totally agree with the consistency.

**Comment by vishalmehta1991** [comment]
@JohnZed I would also recommend n_streams=8/10 for cumlDaskRF

File: cuml/random_forest_demo_mnmg.ipynb [hunk 1]
```diff
@@ -0,0 +1,238 @@
+{
```
Comment (cjnolet) [inline cuml/random_forest_demo_mnmg.ipynb]: Just for consistency, should we use the suffix `_mnmg_demo` or `_demo_mnmg`? I used `_mnmg_demo` for kmeans but I don't mind changing it. Would like to be consistent, though.

File: cuml/random_forest_demo_mnmg.ipynb [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Random Forests Multi-node, Multi-GPU demo\n",
+    "\n",
+    "The experimental cuML Multi-node, Multi-GPU (MNMG) implementation of Random Forests leverages Dask to do embarrassingly-parallel model fitting. For a random forest with N trees being fit by W workers, each worker wil build N / W trees. During inference, predictions from all N trees will be combined.\n",
```
Comment (cjnolet) [inline cuml/random_forest_demo_mnmg.ipynb]:
```suggestion
    "The experimental cuML multi-node, multi-GPU (MNMG) implementation of random forests leverages Dask to do embarrassingly-parallel model fitting. For a random forest with `N` trees being fit by `W` workers, each worker will build `N / W` trees. During inference, predictions from all `N` trees will be combined.\n",
```

File: cuml/random_forest_demo_mnmg.ipynb [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    "\n",
+    "The caller is responsible for partitioning the data efficiently via Dask. To build an accurate model, it's important to ensure that each worker has a representative chunk of the data. This can come by distributing the data evenly after ensuring that it is well shuffled. Or, given sufficient memory capacity, the caller can replicate the data to all workers. This approach will most closely simulate the single-GPU building approach.\n",
+    "\n",
+    "**Note:** cuML 0.9 contains the first, experimental preview release of the MNMG Random Forest model. The API is subject to change in future releases, and some known limitations remain:\n",
```
Comment (cjnolet) [inline cuml/random_forest_demo_mnmg.ipynb]:
```suggestion
    "**Note:** cuML 0.9 contains the first, experimental preview release of the MNMG random rorest model. The API is subject to change in future releases, and some known limitations remain.\n",
```

File: cuml/random_forest_demo_mnmg.ipynb [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    "n_workers = len(workers)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define Parameters\n",
+    "\n",
+    "In addition to the number of examples, Random Forest fitting performance depends heavily on the number of columns in a dataset and (especially) on the maximum depth to which trees are allowed to grow. Lower max_depth values can greatly speed up fitting, though going too low may reduce accuracy."
```
Comment (cjnolet) [inline cuml/random_forest_demo_mnmg.ipynb]:
```suggestion
    "In addition to the number of examples, random forest fitting performance depends heavily on the number of columns in a dataset and (especially) on the maximum depth to which trees are allowed to grow. Lower `max_depth` values can greatly speed up fitting, though going too low may reduce accuracy."
```

File: cuml/random_forest_demo_mnmg.ipynb [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    "n_trees = 1000"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate Data on host\n",
+    "\n",
+    "In this case, we generate data on the client (initial process) and pass it to the workers. You could also load data directly onto the workers via, for example, `dask_cudf.read_csv`. See also the KMeans MNMG notebook for an alternative method of generating data on the worker nodes."
```
Comment (cjnolet) [inline cuml/random_forest_demo_mnmg.ipynb]:
```suggestion
    "In this case, we generate data on the client (initial process) and pass it to the workers. You could also load data directly onto the workers via, for example, `dask_cudf.read_csv()`. See also the k-means MNMG notebook (kmeans_mnmg_demo.ipynb) for an alternative method of generating data on the worker nodes."
```

**Comment by cjnolet** [comment]
> I still don't love that we use ## instead of # for headings, 

I've been using the `#` for the main heading of the notebook, `##` for the subsections (like steps to direct the notebook execution), and `###` for breaking down large steps into smaller sections.

I've been approaching it similarly to how I'd approach a `README.md` or a Wiki post. 

I'm not at all married to this design choice, but consistency is nice.

**Comment by JohnZed** [comment]
@cjnolet - I renamed to random_forest_mnmg_demo to be consistent and added Vishal's streams suggestion in addition to your wording fixes.

Should be good to merge, @taureandyernv !

**Pull request merged**